pkg/helm/urlwatcher.go

```diff
@@ -49,33 +49,37 @@ const (
 
 // checkUpdate checks a SHA URL to determine if the installation package has been updated
 // and fetches the new version if necessary.
-func (p *Poller) checkUpdate() error {
+func (p *Poller) checkUpdate() (bool, error) {
 	uf := p.urlFetcher
 	shaF, err := uf.fetchSha()
 	if err != nil {
-		return err
+		return false, err
 	}
 	hashAll, err := ioutil.ReadFile(shaF)
 	if err != nil {
-		return fmt.Errorf("failed to read sha file: %s", err)
+		return false, fmt.Errorf("failed to read sha file: %s", err)
 	}
 	// Original sha file name is formatted with "HashValue filename"
 	newHash := strings.Fields(string(hashAll))[0]
 
 	if !strings.EqualFold(newHash, p.existingHash) {
 		p.existingHash = newHash
-		return uf.fetchChart(shaF)
+		return true, uf.fetchChart(shaF)
 	}
-	return nil
+	return false, nil
 }
 
-func (p *Poller) poll() {
+func (p *Poller) poll(notify chan struct{}) {
```

Comment:
better to pass in a write-only channel: `notify chan<- struct{}`

Reply:
Also, I recommend creating a channel and returning it from this function.